pages/login.vue

```diff
@@ -0,0 +1,78 @@
+<script setup lang="ts">
+import type { FormError, FormSubmitEvent } from '#ui/types';
+
+const supabase = useSupabaseClient();
+const user = useSupabaseUser();
+
+watchEffect(() => {
+  if (user.value) {
+    console.log('logged in');
+    navigateTo('/today');
+  }
+});
+
+// EMAIL LOGIN setup
+/*
+const state = reactive({
+  email: undefined,
+  password: undefined,
+});
+
+const onEmailLogin = async (event: FormSubmitEvent<any>) => {
+  if (!state.email || !state.password) {
+    return;
+  }
+  const { data, error } = await supabase.auth.signInWithPassword({
+    email: state.email,
+    password: state.password,
+  });
+};
+
+const validate = (state: any): FormError[] => {
+  const errors = [];
+  if (!state.email) errors.push({ path: 'email', message: 'Required' });
+  if (!state.password) errors.push({ path: 'password', message: 'Required' });
+  return errors;
+};
+*/
+</script>
+
+<template>
+  <div class="min-h-full flex flex-col justify-center py-12 sm:px-6 lg:px-8">
+    <h2 class="my-6 text-center text-3xl font-extrabold u-text-white">
+      Sign in to your account
+    </h2>
+    <LoginCard>
+      <!-- <UForm :state="state" :validate="validate" @submit="onEmailLogin">
+        <UFormGroup label="Email" name="email">
+          <UInput v-model="state.email" />
+        </UFormGroup>
+        <UFormGroup label="Password" name="password">
+          <UInput v-model="state.password" type="password" />
+        </UFormGroup>
+        <UButton type="submit"> Login </UButton>
+      </UForm>
+      <UButton
+        class="mt-3"
+        icon="i-mdi-github"
+        block
+        label="Github"
+        @click="supabase.auth.signInWithOAuth({ provider: 'github' })"
+      /> -->
+      <UButton
+        class="mt-3"
+        icon="i-mdi-google"
+        block
+        label="Google"
+        @click="
+          supabase.auth.signInWithOAuth({
+            provider: 'google',
+            options: {
+              redirectTo: 'http://localhost:3000/confirm',
```

Comment:
We cannot use `localhost` for production, so we need to do something about it. One way to deal with it is, to have an [environment variable](https://nuxt.com/docs/api/composables/use-runtime-config#environment-variables), like... 

```
MANYADAY_DOMAIN=localhost:3000
```

↑ The local `.env` file, and you can put the real domain (probably `something.vercel.app` at this point) in your hosting platform's env var configuration.

Reply:
I see - to confirm, I will have to create a .env variable for the development mode url (localhost) and once I deploy to a real domain such as vercel, I input/configure the url on the platform settings -- is this correct?

Reply:
Yes! And now that I think again, MANYADSY_DOMAIN sounds like a wrong name. It could be 

MANYADAY_URL=http://localhost:3000

Reply:
The redirect post google oauth wasn't working after I tried adding the baseURL to the `.env` file. I did a bit of digging and this is what I found: 

https://nuxt.com/docs/guide/going-further/runtime-config
![CleanShot 2024-02-03 at 09 57 54@2x](https://github.com/boyeonihn/manyaday/assets/95451902/ded966f6-fff3-495b-a8af-978430e632b5)
- the docs talk about configuring the runtime configuration in the nuxt.config.ts file, so that's what I ended up trying and now everything seems to work. 

I also looked at the [sample todo nuxt/supabase project codebase](https://github.com/nuxt-modules/supabase/blob/main/demo/nuxt.config.ts) that I referred to for my sample nuxt project and they also did something similar to the docs mentioned above: 

![CleanShot 2024-02-03 at 09 56 56@2x](https://github.com/boyeonihn/manyaday/assets/95451902/3e763f17-e2ac-40e3-b24a-79f877010949)

Reply:
Good job @boyeonihn !

Environment variables often include confidential informations, so frameworks like Nuxt or Next.js has their way to take the env vars into their system.